mllib/src/test/scala/org/apache/spark/mllib/clustering/KMeansSuite.scala

```diff
@@ -257,6 +258,54 @@ class KMeansSuite extends FunSuite with MLlibTestSparkContext {
       assert(predicts(0) != predicts(3))
     }
   }
+
+  test("model save/load") {
+    val tempDir = Utils.createTempDir()
+    val path = tempDir.toURI.toString
+
+    Array(true, false).foreach { case selector =>
+      val model = KMeansSuite.createModel(10, 3, selector)
+      // Save model, load it back, and compare.
+      try {
+        model.save(sc, path)
+        val sameModel = KMeansModel.load(sc, path)
+        KMeansSuite.checkEqual(model, sameModel, selector)
+      } finally {
+        Utils.deleteRecursively(tempDir)
+      }
+    }
+  }
+}
+
+object KMeansSuite extends FunSuite {
+  def createModel(dim: Int, k: Int, isSparse: Boolean): KMeansModel = {
+    val singlePoint = isSparse match {
+      case true =>
+        Vectors.sparse(dim, Array.empty[Int], Array.empty[Double])
+      case _ =>
+        Vectors.dense(Array.fill[Double](dim)(0.0))
+    }
+    new KMeansModel(Array.fill[Vector](k)(singlePoint))
+  }
+
+  def checkEqual(a: KMeansModel, b: KMeansModel, isSparse: Boolean): Unit = {
+    assert(a.k === b.k)
+    isSparse match {
+      case true =>
```

Comment:
I suggest removing `isSparse` and using the following:

~~~scala
a.clusterCenters.zip(b.clusterCenters).foreach {
  case (ca: SparseVector, cb: SparseVector) =>
    assert(ca === cb)
  case (ca: DenseVector, cb: DenseVector) =>
    assert(ca === cb)
  case _ =>
    throw ...
}
~~~